dthm4kaiako/static/scss/_events.scss

```diff
@@ -161,10 +161,162 @@ $declined-dark: rgb(255, 67, 67);
+}
+
+// For Event Details template
+.event-ended-label {
+  justify-content: center;
+  display: flex;
+  background-color:$closed-registrations-or-ended;
+  color:black;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
+
+// For Event Details template
+.event-cancelled-label {
+  justify-content: center;
+  display: flex;
+  background-color:$cancelled-event-or-declined-registration;
+  color:black;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
+
+// For Manage Event Details template
+.not-published-event-status-label {
+  background-color:$not-published-event-or-pending-registration;
+  color:black;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
+
+// For Manage Event template
+.published-event-status-label {
+  background-color:$published-event-or-approved-registration;
+  color:black;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
+
+// For Manage Event template
+.cancelled-event-status-label {
+  background-color:$cancelled-event-or-declined-registration;
+  color:black;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
+
+// For Manage Event template
+.publish-event-btn {
+  color:white;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 10px;
+  padding-bottom: 10px;
+}
+
+// For Manage Event template
+.cancel-event-btn {
+  color:white;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 10px;
+  padding-bottom: 10px;
+}
+
+// For Manage Event template
+.view-public-page-btn {
+  color:white;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 10px;
+  padding-bottom: 10px;
+}
+
+// For Manage Event Registration template
+.approved-status-label {
+  background-color:$published-event-or-approved-registration;
+  color:black;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
+
+// For Manage Event Registration template
+.pending-status-label {
+  background-color:$not-published-event-or-pending-registration;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
+
+// For Manage Event Registration template
+.declined-status-label {
+  background-color:$cancelled-event-or-declined-registration;
+  font-size: medium;
+  border-radius: 8px;
+  padding-left: 12px;
+  padding-right: 12px;
+  padding-top: 12px;
+  padding-bottom: 12px;
+}
```

Comment:
This could be simplified as:

```suggestion
// Abstract classes
%event-label {
    font-size: medium;
    border-radius: 8px;
    padding-left: 12px;
    padding-right: 12px;
    padding-top: 12px;
    padding-bottom: 12px;
}
%event-button {
    border-radius: 8px;
    padding-left: 12px;
    padding-right: 12px;
    padding-top: 10px;
    padding-bottom: 10px;
}

// For Event Details template
.closed-registrations-label,
.event-ended-label {
  justify-content: center;
  display: flex;
  background-color:$closed-registrations-or-ended;
  color:black;
  @extend %event-label;
}
.event-cancelled-label {
  justify-content: center;
  display: flex;
  background-color:$cancelled-event-or-declined-registration;
  color:black;
  @extend %event-label;
}

// For Manage Event Details template
.not-published-event-status-label {
  background-color:$not-published-event-or-pending-registration;
  color:black;
  @extend %event-label;
}

// For Manage Event template
.published-event-status-label {
  background-color:$published-event-or-approved-registration;
  color:black;
  @extend %event-label;
}

// For Manage Event template
.cancelled-event-status-label {
  background-color:$cancelled-event-or-declined-registration;
  color:black;
  @extend %event-label;
}

// For Manage Event template
.publish-event-btn, 
.cancel-event-btn, 
.view-public-page-btn {
  color:white;
  @extend %event-button;
}

// For Manage Event Registration template
.approved-status-label,
.pending-status-label, 
.declined-status-label {
  @extend %event-label;
}
.approved-status-label {
  background-color:$published-event-or-approved-registration;
  color:black;
}
.pending-status-label {
  background-color:$not-published-event-or-pending-registration;
}
.declined-status-label {
  background-color:$cancelled-event-or-declined-registration;
}
```

It's not ideal since you are repeating a lot of generated CSS, it would be better to have multiple classes per HTML element, so you could change the abstract classes into a normal class to the elements have multiple styles applied. However it's fine for this PR.

Reply:
Updated